src/main/java/pl/mmieczak/controller/MainController.java

```diff
@@ -0,0 +1,86 @@
+package pl.mmieczak.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import pl.mmieczak.model.Category;
+import pl.mmieczak.model.Product;
+import pl.mmieczak.model.ProductRepository;
+import pl.mmieczak.utils.Utilities;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Controller
+public class MainController {
+
+    ProductRepository productRepository;
+
+    public MainController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String main(Model model) {
+        return "index";
+    }
+
+    @GetMapping("/lista")
+    public String products(Model model, @RequestParam("kategoria") String category) {
+        List<Product> productList = productRepository.getAll();
+        BigDecimal sumOfProductPrices;
+
+        switch (category) {
```

Comment:
Zauważ że jakbyś miał od razu `enum` zamiast Stringa to mógłbyś się całkowicie pozbyć tego switcha. Wywołujesz w nim zawsze tę samą metodę a różni się tylko enum.